src/Tools/dotnet-monitor/HostBuilderHelper.cs

```diff
@@ -65,37 +67,51 @@ public static IHostBuilder CreateHostBuilder(string[] urls, string[] metricUrls,
                 .UseContentRoot(AppContext.BaseDirectory) // Use the application root instead of the current directory
                 .ConfigureHostConfiguration((IConfigurationBuilder builder) =>
                 {
-                    //Note these are in precedence order.
-                    ConfigureEndpointInfoSource(builder, diagnosticPort);
-                    ConfigureMetricsEndpoint(builder, metrics, metricUrls);
-                    ConfigureGlobalMetrics(builder);
-                    builder.ConfigureStorageDefaults();
+                    if (TestingMode == ConfigurationTestingMode.None)
+                    {
+                        //Note these are in precedence order.
+                        ConfigureEndpointInfoSource(builder, diagnosticPort);
+                        ConfigureMetricsEndpoint(builder, metrics, metricUrls);
+                        ConfigureGlobalMetrics(builder);
+                        builder.ConfigureStorageDefaults();
 
-                    builder.AddCommandLine(new[] { "--urls", ConfigurationHelper.JoinValue(urls) });
+                        builder.AddCommandLine(new[] { "--urls", ConfigurationHelper.JoinValue(urls) });
+                    }
                 })
                 .ConfigureAppConfiguration((IConfigurationBuilder builder) =>
                 {
+                    if (TestingMode != ConfigurationTestingMode.None)
```

Comment:
For context, this is the area I've been needing @jander-msft to take a look at; we had previously discussed a way to get rid of some of these checks regarding testing, but in the meantime they're still here and I'm not sure if we resolved what we were going to do about them. If we can get rid of them, then we shouldn't need to pass around `TestingMode`